File: 1290 - Convert Binary Number in a Linked List to Integer/cpp/ans.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    // method 1
    int getDecimalValue(ListNode* head) {
        ListNode* ptr = head;
        vector<int> st;
        while(ptr){
            st.push_back(ptr->val);
            ptr = ptr->next;
        }
        // convert to binary
        int result = 0;
        for(auto x: st) {
            result <<= 1;
            result += x;
        }
        return result;
    }
    // method 2 
    int getDecimalValue(ListNode* head) {
        ListNode* ptr = head;
        int result = 0;
        while(ptr){
            result = result << 1;
            result += ptr->val;
            ptr = ptr->next;
        }
        
        return result;
    }
};

File: 1290 - Convert Binary Number in a Linked List to Integer/go/ans.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 // method 1
 func getDecimalValue(head *ListNode) int {
    ptr := head
	var st []int
	for ptr != nil {
		st = append(st, ptr.Val)
		ptr = ptr.Next
	}

	ret := 0
	for _, x := range st {
		ret <<= 1
		ret += x
	}

	return ret
}
 // method 2
func getDecimalValue(head *ListNode) int {
    ptr := head
	ret := 0
	for ptr != nil {
		ret <<= 1
		ret += ptr.Val
		ptr = ptr.Next
	}

	return ret
}